var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -0,0 +1,154 @@
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Qmcpack(CMakePackage):
+    """QMCPACK, is a modern high-performance open-source Quantum Monte 
+       Carlo (QMC) simulation code."""
+
+    # Package information
+    homepage = "http://www.qmcpack.org/"
+    url      = "https://github.com/QMCPACK/qmcpack/archive/v3.1.0.tar.gz"
+
+    version('3.1.0', 'bdf3acd090557acdb6cab5ddbf7c7960')
+    version('3.0.0', '75f9cf70e6cc6d8b7ff11a86340da43d')
+
+    # This download method is untrusted, and is not recommend by th Spack manual.
+    version('develop', git='https://github.com/QMCPACK/qmcpack.git')
+
+    # These defaults match those in the QMCPACK manual
+    variant('debug', default=False, description='Build debug version')
+    variant('mpi', default=True, description='Build with MPI support')
+    variant('cuda', default=False, description='Enable CUDA and GPU acceleration.')
+    variant('complex', default=True, description='Build the complex (general twist/k-point) version')
+    variant('mixed', default=False, description='Build the mixed precision (mixture of single and double precision) version for gpu and cpu (experimental)')
+    variant('gui', default=False, description='Install with Py-Matplotlib (long installation time)') 
+
+    depends_on('cmake@3.4.3:', type='build')
+    depends_on('mpi', when='+mpi')
+    depends_on('libxml2')
+    depends_on('hdf5+mpi', when='+mpi')
+    depends_on('hdf5~mpi', when='~mpi')
+    depends_on('boost')
+    depends_on('blas')
+    depends_on('lapack')
+    depends_on('fftw+mpi', when='+mpi')
+    depends_on('fftw~mpi', when='~mpi')
+    depends_on('cuda', when='+cuda')
+   
+    # qmcpack needs these for data analysis
+    depends_on('python', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+
+    # GUI is optional and leads to a long complex DAG for dependencies
+    depends_on('py-matplotlib', type=('build', 'run'), when='+gui')
+
+    # B-spline basis calculation require a patched version of Quantum Espresso 5.3.0 (see QMCPACK manual)
+    depends_on('espresso@5.3.0+qmcpackconv~elpa', when='+mpi')
+    depends_on('espresso@5.3.0+qmcpackconv~elpa~scalapack~mpi', when='~mpi')
+
+    def cmake_args(self):
+        args = []
+
+        filter_file(r'$ENV{LIBXML2_HOME}/lib',
+                    '${LIBXML2_HOME}/lib $ENV{LIBXML2_HOME}/lib',
+                    'CMake/FindLibxml2QMC.cmake')
+  
+        print dir(self.spec['hdf5'].prefix)
+        print self.spec['boost'].prefix
+
+        if 'cxxflags' in self.compiler.flags:
+            cxx_flags = ' '.join(self.compiler.flags['cxxflags'])
+            args.append('-DCMAKE_CXX_FLAGS={0}'.format(cxx_flags))
+
+        if 'cflags' in self.compiler.flags:
+            c_flags = ' '.join(self.compiler.flags['cflags'])
+            args.append('-DCMAKE_C_FLAGS={0}'.format(c_flags))
```

Comment:
As of the time of the pull request, CMakePackage wasn't passing the flags forward. I wasn't entirely sure if that was a bug or not. 

FWIW, the developers of qmcpack provide their own logic for deciding on processor flags that includes the equivalent of `-march=native` in `CMake/${Compiler}.cmake` that are possibly bad ideas in multi-architecture environments, but the Intel version at least is smart enough to not override Spack if `CMAKE_C_FLAGS` includes `-x` or `-ax` and is cumulative for GNU.